File: v1-nestjs/server/src/personal/personal.controller.ts
import {
  Controller,
  Get,
  Post,
  Patch,
  Delete,
  Param,
  Body,
  Query,
} from '@nestjs/common';
import { PersonalService } from './personal.service';
import { CreatePersonalFileDto } from './dto/create-personal-file.dto';

@Controller('personal')
export class PersonalController {
  constructor(private readonly personalService: PersonalService) {}

  @Get('files')
  async findAll(@Query('userId') userId: string) {
    const files = await this.personalService.findAll(userId);
    return { code: 0, data: files };
  }

  @Post('files')
  async create(@Body() body: CreatePersonalFileDto) {
    const file = await this.personalService.create(body);
    return { code: 0, data: file };
  }

  @Delete('files/:id')
  async remove(@Param('id') id: string) {
    const file = await this.personalService.remove(id);
    return { code: 0, data: file };
  }

  // === Notes ===

  @Post('notes')
  async createNote(@Body() body: { userId: string; title: string; content: string; category?: string }) {
    const note = await this.personalService.createNote(body.userId, body.title, body.content, body.category);
    return { code: 0, data: note };
  }

  @Get('notes')
  async getNotes(
    @Query('userId') userId: string,
    @Query('category') category?: string,
    @Query('page') page?: string,
    @Query('pageSize') pageSize?: string,
  ) {
    const result = await this.personalService.getNotes(userId, {
      category,
      page: page ? parseInt(page, 10) : undefined,
      pageSize: pageSize ? parseInt(pageSize, 10) : undefined,
    });
    return { code: 0, data: result };
  }

  @Patch('notes/:id')
  async updateNote(@Param('id') id: string, @Body() body: { title?: string; content?: string; category?: string }) {
    const note = await this.personalService.updateNote(id, body);
    return { code: 0, data: note };
  }

  @Delete('notes/:id')
  async removeNote(@Param('id') id: string) {
    const note = await this.personalService.removeNote(id);
    return { code: 0, data: note };
  }
}

File: syncflow-java/syncflow-common/src/main/java/com/syncflow/common/controller/personal/PersonalController.java
package com.syncflow.common.controller.personal;

import com.syncflow.common.dto.personal.*;
import com.syncflow.common.result.PageResult;
import com.syncflow.common.result.Result;
import com.syncflow.common.service.personal.PersonalService;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * Personal workspace controller (files + notes).
 */
@RestController
@RequestMapping("/api/personal")
public class PersonalController {

    private final PersonalService personalService;

    public PersonalController(PersonalService personalService) {
        this.personalService = personalService;
    }

    /**
     * Paginated personal files.
     */
    @GetMapping("/files")
    public Result<PageResult<PersonalFileVO>> getPersonalFiles(
            @RequestParam(defaultValue = "1") int pageNum,
            @RequestParam(defaultValue = "10") int pageSize) {
        PageResult<PersonalFileVO> result = personalService.getPersonalFiles(pageNum, pageSize);
        return Result.success(result);
    }

    /**
     * Create a personal file entry.
     */
    @PostMapping("/files")
    public Result<PersonalFileVO> createPersonalFile(@Valid @RequestBody CreatePersonalFileDTO dto) {
        PersonalFileVO vo = personalService.createPersonalFile(dto);
        return Result.success(vo);
    }

    /**
     * Delete a personal file.
     */
    @DeleteMapping("/files/{id}")
    public Result<Void> deletePersonalFile(@PathVariable Long id) {
        personalService.deletePersonalFile(id);
        return Result.success();
    }

    /**
     * Paginated notes.
     */
    @GetMapping("/notes")
    public Result<PageResult<NoteVO>> getNotes(
            @RequestParam(required = false) String keyword,
            @RequestParam(defaultValue = "1") int pageNum,
            @RequestParam(defaultValue = "10") int pageSize) {
        PageResult<NoteVO> result = personalService.getNotes(keyword, pageNum, pageSize);
        return Result.success(result);
    }

    /**
     * Create a note.
     */
    @PostMapping("/notes")
    public Result<NoteVO> createNote(@Valid @RequestBody CreateNoteDTO dto) {
        NoteVO vo = personalService.createNote(dto);
        return Result.success(vo);
    }

    /**
     * Update a note.
     */
    @PatchMapping("/notes/{id}")
    public Result<NoteVO> updateNote(@PathVariable Long id,
                                     @Valid @RequestBody CreateNoteDTO dto) {
        NoteVO vo = personalService.updateNote(id, dto);
        return Result.success(vo);
    }

    /**
     * Delete a note.
     */
    @DeleteMapping("/notes/{id}")
    public Result<Void> deleteNote(@PathVariable Long id) {
        personalService.deleteNote(id);
        return Result.success();
    }
}
